src/cabin/cabin.utils.ts

```diff
@@ -0,0 +1,30 @@
+export function dateIsValid(dateString: string): boolean {
+  const regex = /^\d{4}-\d{2}-\d{2}$/;
+
+  if (dateString.match(regex) === null) {
+    return false;
+  }
+
+  const date = new Date(dateString);
+  const timestamp = date.getTime();
+
+  if (typeof timestamp !== 'number' || Number.isNaN(timestamp)) {
+    return false;
+  }
+
+  return date.toISOString().startsWith(dateString);
+}
+
+export function getVisbookId(url: string): number {
+  const urlParts = url.split('/');
+  for (const part of urlParts) {
+    const int = parseInt(part, 10);
+    if (isNaN(int)) {
```

Comment:
If the part is an empty string this will also result in a zero being returned. Perhaps they should be skipped?

Reply:
I'm not sure if that's the case? There is always an empty string, since most (all?) booking urls contain //.
Here are the iterations I get from one example:
```
Part 1: "https:" -> NaN
Part 2: "" -> NaN
Part 3: "reservations.visbook.com" -> NaN
Part 4: "6260?lang=no" -> 6260
```
I guess I'm lucky that it just ignores the rest of part 4, but the empty string in part 2 is NaN and therefore skipped.